src/bos/server/migrations/db.py

```diff
@@ -33,6 +34,35 @@
 STAT_DB = dbutils.get_wrapper(db='session_status')
 COMP_DB = dbutils.get_wrapper(db='components')
 
+MAX_DB_WAIT_SECONDS=120.0
+
+def all_db_ready() -> bool:
+    """
+    Wait for up to MAX_DB_WAIT_SECONDS for all databases to be ready
+    """
+    start_time = time.time()
+    first = True
+    while time.time() - start_time <= MAX_DB_WAIT_SECONDS:
+        if first:
+            first = False
+        else:
+            LOGGER.info("Sleeping for 7 seconds before retrying databases")
+            time.sleep(7)
+        if not TEMP_DB.ready:
+            continue
+        LOGGER.info("Template database is ready")
```

Comment:
Don't we just need the template database to be ready? We migrating V1 session templates over to V2 session templates, so they're being migrated from the etcd database to the redis database. If I'm remembering correctly, it doesn't involve any other databases, so checking them is unnecessary.

One reason that is important is that Helm will time out after five minutes, so a delay that lasts that long will scuttle the deployment.

Reply:
No, we use all 4 of those databases. The migration code for CSM 1.6 does checks to make sure the data is not illegal per the spec.

Reply:
But if the databases aren't ready, the deployment will fail one way or the other, because the migration job will barf once it tries to access them

Reply:
If the default 5m helm timeout is an issue, it can be changed in the manifest to a longer value.

Reply:
It's actually 10 minutes in the manifest, I'm pretty sure. But timeouts have not been a problem. The databases have always become ready pretty quickly. But they are not always ready immediately when the job starts to run, hence this PR, to avoid ugly failures that would go away if we just retried.

Reply:
And to be clear, the problem here is not that the deployment fails. The job is automatically retried if the databases aren't ready, and it usually succeeds on the second try. But it leaves behind an ugly failed pod which could alarm administrators needlessly.